### PR TITLE
Feature/josecaste jwt migration

### PR DESCRIPTION
**Pull request by JoseCaste**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:8-jdk-alpine
+COPY  "./target/sport-hub.jar" "sport-hub-docker-ex.jar"
+EXPOSE 8080
+#ENTRYPOINT ["java", "-jar", "-Dbucket_name='sport-hub-11400.appspot.com' -Demail_pwd='12345' -Demail_username='jjcastellanosguzman@outlook.es' -Dproject_id=sport-hub-11400 -Dservice_account='D:\IT Academy\Couse Docs\sport-hub-11400-firebase-adminsdk-pxvmd-2bc8df8b47.json' -Duser_db=postgres -Duser_pwd=cGJZ@z3#B@xq9JM", "sport-hub-docker-ex.jar"]
```
Comment (cesar-martinezg) [inline Dockerfile]: remove unused code

File: pom.xml [hunk 2]
```diff
@@ -86,7 +88,7 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-tomcat</artifactId>
-			<scope>provided</scope>
+			<!--<scope>provided</scope>-->
```
Comment (cesar-martinezg) [inline pom.xml]: remove if not used OR uncomment code if needed

File: src/main/java/com/softserve/api/auth/AuthRestController.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        return ResponseEntity.ok()
+                .body(new UserInfoResponse(userOptional.get().getIdAppUser(),
+                        userOptional.get().getFirstName(),
+                        userOptional.get().getEmail(),
+                        roles,
+                        token));
+    }
+
+    /*@PostMapping("/signup")
```
Comment (cesar-martinezg) [inline src/main/java/com/softserve/api/auth/AuthRestController.java]: remove unused code

File: src/main/java/com/softserve/dto/LoginRequest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class LoginRequest {
+
+    private String userName;
```
Comment (cesar-martinezg) [inline src/main/java/com/softserve/dto/LoginRequest.java]: "username" is a single word, no need for upper casing the N letter

File: src/main/java/com/softserve/dto/UserInfoResponse.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserInfoResponse {
+
+    private Integer id;
+
+    private String userName;
```
Comment (cesar-martinezg) [inline src/main/java/com/softserve/dto/UserInfoResponse.java:17]: username is a single word, not needed to use uppercase for the N

File: src/main/java/com/softserve/security/user/UserDetailsServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    @Autowired
+    IUserRepository userRepository;
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User Not Found with username: " + username));
```
Comment (cesar-martinezg) [inline src/main/java/com/softserve/security/user/UserDetailsServiceImpl.java:22]: is this exception message passed to the UI?

If your answer is yes, I would advise not doing it, usually, when talking about security during login, we just respond to the user/client saying "username and/or password is incorrect", meaning, we do not facilitate to a potential attacker which of the fields is incorrect or does not exist.

**Review by cesar-martinezg — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/security/user/UserDetailsServiceImpl.java (hunk 6 above)
Comment (JoseCaste) [reply src/main/java/com/softserve/security/user/UserDetailsServiceImpl.java:22]: Ow, nope, sorry, this exception is not passed to UI, this error only can be see from log files.

**Review by cesar-martinezg — DISMISSED** [review]

**Review by cesar-martinezg — APPROVED** [review]

**Pull request merged**